sql/core/src/test/resources/sql-tests/results/group-by-filter.sql.out

```diff
@@ -796,14 +796,16 @@ IN/EXISTS predicate sub-queries can only be used in Filter/Join and a few comman
 :     +- Filter (dept_id#x = outer(dept_id#x))
 :        +- SubqueryAlias dept
 :           +- View (`DEPT`, [dept_id#x,dept_name#x,state#x])
-:              +- Project [dept_id#x, dept_name#x, state#x]
-:                 +- SubqueryAlias DEPT
-:                    +- LocalRelation [dept_id#x, dept_name#x, state#x]
+:              +- Project [cast(dept_id#x as int) AS dept_id#x, cast(dept_name#x as string) AS dept_name#x, cast(state#x as string) AS state#x]
```

Comment:
There are some newly added `cast`. Are they redundant?

Reply:
Redundant casts in an analyzing phase looks fine to me.

Reply:
Can we add a rule at the end of `Analyzer` if the plan is resolved to check the top `Project` and reduce the`Cast` if is redundant ? The UpCast seems to avoid the table reference changed before view analysis but we can remove it after analysis.

Reply:
It's better to delay the cast adding (after the parsed view plan is resolved), so that we can skip adding cast for views that have no schema changing. But I can't find an easy way to do it and this is really not a big deal (optimizer willl remove redundant casts), so I go with the simple approach for maintainability.

Reply:
sounds okay.